pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1336,6 +1336,19 @@ func parseIpspec(ipspec *zconfig.Ipspec,
 				s, err))
 		}
 		config.Subnet = *subnet
+		// set dhcp range in the second half of the subnet(for larger subnets),
+		// set the first address in the dhcp range as gateway
+		prefixLen, _ := subnet.Mask.Size()
+		prefixRange := 0x01 << (32 - prefixLen)
+		start := 1
+		end := prefixRange - 3
+		if prefixRange >= types.LargeAppCount {
+			start = prefixRange / 2
+			end = prefixRange/2 - 3
+		}
+		config.Gateway = types.AddToIP(subnet.IP, start)
+		config.DhcpRange.Start = types.AddToIP(config.Gateway, 1)
+		config.DhcpRange.End = types.AddToIP(config.DhcpRange.Start, end)
```

Comment:
I swear I had typed up a comment on this, saying that we don't want to do this at the top but do it further down so that we can log the case when we need to derive these defaults (as log.Warnf) when DhcpRange and Gateway respectively were not set. That means in two places since one of them might be set and not the other.